.github/workflows/bench-benchmark.yml

```diff
@@ -0,0 +1,142 @@
+        with:
+          terraform: true
+          k3s: true
+          docker: true
+          aws: true
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_REGION: eu-west-3
+
+      - name: Get core version
+        run: |
+          set -ex
+          echo "core-version=$(cat versions.tfvars.json | jq -r '.armonik_versions.core')" >> $GITHUB_ENV
+
+      - id: deploy
+        name: "Deploy ArmoniK"
+        uses: aneoconsulting/ArmoniK.Action.Deploy/deploy@main
+        with:
+          type: ${{ matrix.type }}
+          prefix: ${{ env.prefix }}
+          core-version: ${{ env.core-version }}
+          parameters-file-path: ${{ env.parameters-file-path }}
+
+      - id: get-armonik-endpoint
+        name: "Get ArmoniK's control plane endpoint"
+        env:
+          TYPE: ${{ matrix.type }}
+        run: |
+          set -ex
+          grpc_endpoint=$(cat "infrastructure/quick-deploy/$TYPE/generated/armonik-output.json" | jq -r '.armonik.control_plane_url' | sed -r 's/(http:\/\/)([^:]*)(:.*)/\2/')
+          echo "grpc-endpoint=$grpc_endpoint" >> "$GITHUB_OUTPUT"
+          sleep 5
+
+      - id: bench
+        name: Run Bench
+        uses: aneoconsulting/ArmoniK.Action.Deploy/bench@main
+        with:
+          type: ${{ matrix.type }}
+          armonik-core-version: ${{ env.core-version }}
+          ntasks: ${{ matrix.ntasks }}
+          session-name: bench
+          grpc-client-endpoint: ${{ steps.get-armonik-endpoint.outputs.grpc-endpoint }}
+          timeout: 1200
+
+      - id: get-bench-stats
+        name: Get Bench Stats
+        uses: aneoconsulting/ArmoniK.Action.Deploy/get-throughput@main
+        with:
+          grpc-client-endpoint: ${{ steps.get-armonik-endpoint.outputs.grpc-endpoint }}
+          session-name: ${{ steps.bench.outputs.session-name }}
+          poll-duration-limit: ${{ matrix.polling-limit }}
+
+      - name: Upload benchmark results to artifact registry
+        uses: actions/upload-artifact@v4
+        with:
+          name: benchclient_benchmark_${{ github.event_name }}_${{ matrix.type }}_${{ github.run_id }}
+          path: ${{ steps.get-bench-stats.outputs.bench-file-path }}
+    
+      - name: Upload benchmark results to s3
+        env:
+          EVENT_NAME: ${{ github.event_name }}
+          BENCH_RESULTS_PATH: ${{ steps.get-bench-stats.outputs.bench-file-path }}
+          REF_NAME: ${{ github.ref_name }}
+          TYPE: ${{ matrix.type }}
+          GHRUNID: ${{ github.run_id }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_EC2_METADATA_DISABLED: true
+        run: |
+          DATE=$(date +"%Y-%m-%d")
+          aws s3 cp "$BENCH_RESULTS_PATH" "s3://armonik-bench-storage/${REF_NAME}/${GHRUNID}_${DATE}/benchclient_benchmark_${EVENT_NAME}_${TYPE}.json"
```

Comment:
I am not sure about REF_NAME. For branches, it is the name of the branch, that's fine, but in case of a release, I would much prefer to have `release` instead of the tag name. This would also mean that we need to add the tag name in the subfolder name.